src/dune_rules/lib.ml

```diff
@@ -1714,9 +1714,21 @@ module DB = struct
         [ ("name", Lib_name.to_dyn name) ]
     | Some lib -> Compile.for_lib resolve ~allow_overlaps t lib
 
-  let resolve_user_written_deps_for_exes t exes ?(allow_overlaps = false)
-      ?(forbidden_libraries = []) deps ~pps ~dune_version ~optional =
+  let resolve_user_written_deps_for_exes t exes ?preprocess
+      ?(allow_overlaps = false) ?(forbidden_libraries = []) deps ~pps
+      ~dune_version ~optional =
     let lib_deps_info =
+      let pps =
+        match preprocess with
+        | None -> pps
+        | Some pp ->
+          let resolve = resolve t in
+          Preprocess.Per_module.pps
+            (Preprocess.Per_module.with_instrumentation pp
+               ~instrumentation_backend:
+                 (instrumentation_backend ~do_not_fail:true t.instrument_with
+                    resolve))
```

Comment:
Should't this be somehow combined with the `~pps` argument of `resolve_user_written_deps_for_exes` ?

Reply:
You are right. Actually, I just realized that this PR is not necessary. We already pass the correct `pps` in the caller:

```
  let pps =
    Preprocess.Per_module.pps
      (Preprocess.Per_module.with_instrumentation exes.buildable.preprocess
         ~instrumentation_backend:
           (Lib.DB.instrumentation_backend (Scope.libs scope)))
  in
```

It's a little confusing that this step is done inside `Lib` for libraries and in `exe_rules` for executables.

Closing, sorry for the noise.